--- kf.desktop	2005-02-23 20:48:15.000000000 +0100
+++ fix/kf.desktop	2005-02-25 23:59:53.940405904 +0100
@@ -1,15 +1,17 @@
 [Desktop Entry]
-Encoding=UTF-8
 Name=kf jabber client
 Name[pl]=kf klient sieci jabber
 Name[lt]=Pokalbių programa kf
 GenericName=Instant Messenger
 GenericName[pl]=Komunikator Internetowy
 Comment=Instant messaging to people, using Jabber, ICQ, MSN and other clients
+Comment[pl]=Komunikator dla ludzi używających Jabbera, ICQ, MSN i innych klientów
 Comment[lt]=Pokalbiai internete (su žmonėmis, naudojančiais Jabber, ICQ, MSN ir kitus protokolus)
-Exec=kf
+Exec=kfisha
 Icon=kf.png
+StartupNotify=false
 Terminal=false
 Type=Application
-Categories=GTK;Application;Network;
-StartupNotify=true
+Categories=GTK;Network;InstantMessaging;
+Encoding=UTF-8
+# vi: encoding=utf-8
